testsuite/tests/backtrace/filter-locations.sh

```diff
@@ -1,2 +1,5 @@
 #!/bin/sh
-sed -e "s/^.*in file/File/" -e 's/ (inlined)//' -e "s/, raise.*//" | grep ^File
+# This location filter is erasing information from the backtrace
+# to be robust to different inlining choices made by different compiler settings.
+# It checks that the expected locations occur (in the expected order).
+sed -e "s/^.*in file/File/" -e 's/ (inlined)//' | grep ^File
```

Comment:
```suggestion
CR=$(printf "\r")
sed -e "s/^.*in file/File/" -e 's/ (inlined)//' -e "s/, raise[^$CR]*//" | grep ^File
```
You could do it this way, or insert an actual CR character instead of `$CR`. Some versions of `sed` support `\r` as an escape sequence, but I doubt the lowest-denominator does.

Reply:
But actually I'm happy to keep the `, raise` around. If I understand correctly (and the CI successes seem to confirm this), the fragility that the filter protects is the choice of inlining, but all backtraces should agree on which location raised the exception!